ckanext/geocat/utils/xpath_utils.py

```diff
@@ -224,28 +264,33 @@ def xpath_get_distribution_from_distribution_node(
                 description_node)
     else:
         distribution['description'] = {'en': '', 'it': '', 'de': '', 'fr': ''}
-    normed_protocol, protocol_name = _get_normed_protocol(protocol)
-    distribution['protocol'] = normed_protocol
-    distribution['protocol_name'] = protocol_name
+
+    normed_protocol = _get_normed_protocol(protocol)
+    distribution['protocol'] = protocol
+    distribution['normed_protocol'] = normed_protocol
+
     if normed_protocol == DOWNLOAD_PROTOCOL and protocol.startswith(DOWNLOAD_PROTOCOL + ':'):  # noqa
         format = protocol.replace(DOWNLOAD_PROTOCOL + ':', '')
-        media_type = protocol.replace(DOWNLOAD_PROTOCOL + ':', '')
-        distribution['format'] = format
-        distribution['media_type'] = media_type
-    resource_formats = filter(
-        lambda i: i not in [LINKED_DATA_PROTOCOL, MAP_PROTOCOL],
-        SERVICE_PROTOCOLS)
-    if normed_protocol in resource_formats:
+        if format:
+            distribution['media_type'] = format
+            distribution['format'] = format
+        else:
+            distribution['media_type'] = ""
+    elif normed_protocol == DOWNLOAD_PROTOCOL:
+        distribution['media_type'] = ""
+    elif normed_protocol in FORMATED_SERVICE_PROTOCOLS:
         format = re.findall(r'(?<=:).*$', normed_protocol)[0]
         distribution['format'] = format
```

Comment:
The format mapping is slightly changed to avoid setting format or media_type to blank if they are not set, since this is generally a bad style on DCAT and should only be done if necessary. Therefore a download url, where the media type can't be derived still gets a blank media type. For Service protocols only the format is set. For download urls both format and media type are set if the media type can be derived.